src/odemis/acq/align/fastem.py

```diff
@@ -18,22 +20,282 @@
+            self._scanner.blanker.value = True  # always blank the beam to reduce beam damage on sample
+            logging.debug("Finish calibrations.")
+
+    def run_calibrations(self, calibration):
+        """
+        Run a calibration.
+        # Note: All calibrations can be run on bare scintillator.
+        """
+        if calibration == OPTICAL_AUTOFOCUS:
+            autofocus_multiprobe.run_autofocus(self._scanner, self._multibeam, self._descanner, self._detector,
+                                               self._dataflow, self._ccd, self._stage)
+
+        if calibration == SCAN_ROTATION_PREALIGN:
+            scan_rotation_pre_align.run_scan_rotation_pre_align(self._scanner, self._multibeam, self._descanner,
+                                                                self._detector, self._dataflow, self._ccd)
+
+        if calibration == SCAN_AMPLITUDE_PREALIGN:
+            scan_amplitude_pre_align.run_scan_amplitude_pre_align(self._scanner, self._multibeam, self._descanner,
+                                                                  self._detector, self._dataflow, self._ccd)
+
+        if calibration == DESCAN_GAIN_STATIC:
+            descan_gain.run_descan_gain_static(self._scanner, self._multibeam, self._descanner,
+                                               self._detector, self._dataflow, self._ccd)
+
+        if calibration == IMAGE_ROTATION_PREALIGN:
+            image_rotation_pre_align.run_image_rotation_pre_align(self._scanner, self._multibeam, self._descanner,
+                                                                  self._detector, self._dataflow, self._ccd,
+                                                                  self._det_rotator)
+
+        if calibration == IMAGE_TRANSLATION_PREALIGN:
+            image_translation_pre_align.run_image_translation_pre_align(self._scanner, self._multibeam,
+                                                                        self._descanner, self._detector,
+                                                                        self._dataflow, self._ccd)
+
+        if calibration == IMAGE_ROTATION_FINAL:
+            image_rotation.run_image_rotation(self._scanner, self._multibeam, self._descanner,
+                                              self._detector, self._dataflow, self._det_rotator)
+
+        if calibration == IMAGE_TRANSLATION_FINAL:
+            image_translation.run_image_translation(self._scanner, self._multibeam, self._descanner,
+                                                    self._detector, self._dataflow, self._ccd)
+
+    def cancel(self, future):
+        """
+        Cancels the calibrations.
+        :param future: (future) The calibration future.
+        :return: (bool) True if cancelled, TODO False?
+        """
+        self._cancelled = True
+
+        # FIXME Currently there is no subfuture implemented for each calibration. So currently,
+        #  when cancelling the calibration is still finished.
```

Comment:
Can you clarify this sentence a bit more:

`So currently, when cancelling if a calibration has already started it will run until it is completely finished.`